chez-libs/tests/basic.output.md

```diff
@@ -0,0 +1,93 @@
+
+```unison
+generateSchemeBuiltinLibrary _ =
+	fh = open (FilePath "../unison/builtin-generated.ss") Write
+	putText fh (generateBaseFile builtinSpec)
+	close fh
+
+schemeToFile dest link = 
+	fh = open (FilePath dest) Write
+	putText fh (generateScheme false link)
+	close fh
+
+runChez fileName = IO.Process.call "scheme" ["--libdirs", "../", "--script", fileName]
+
+runInScheme id term =
+	fileName = "basic-" ++ (Nat.toText id) ++ ".ss"
+	schemeToFile fileName term
+	runChez fileName
+```
+
+```ucm
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      generateSchemeBuiltinLibrary : ∀ _. _ ->{IO, Exception} ()
+      runChez                      : Text ->{IO} Nat
+      runInScheme                  : Nat
+                                     -> Term
+                                     ->{IO, Exception} Nat
+      schemeToFile                 : Text
+                                     -> Term
+                                     ->{IO, Exception} ()
+
+```
+```ucm
+.> add
+
+  ⍟ I've added these definitions:
+  
+    generateSchemeBuiltinLibrary : ∀ _. _ ->{IO, Exception} ()
+    runChez                      : Text ->{IO} Nat
+    runInScheme                  : Nat
+                                   -> Term
+                                   ->{IO, Exception} Nat
+    schemeToFile                 : Text
+                                   -> Term
+                                   ->{IO, Exception} ()
+
+.> run generateSchemeBuiltinLibrary
+
+  ()
+
+```
+```unison
+test1_term = '(printLine "Hello")
+```
+
+```ucm
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      test1_term : '{IO, Exception} ()
+
+```
+```unison
+test1 = '(runInScheme 1 (termLink test1_term))
+```
+
+```ucm
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      test1 : '{IO, Exception} Nat
+
+```
+```ucm
+.> run test1
+
+  0
```

Comment:
there's something weird about the interaction of transcripts w/ IO.process; the stdout of the IO.Process.call doesn't get captured here, but it shows up fine if you `run` from `ucm`. @dolio do you know what might be happening here?

Reply:
Yeah. Transcripts just don't capture standard output at all. This is also true if you just call prints in unison. I think if you want the output of the process to be displayed in the transcript, you're going to have to use the more complicated API with explicit handles.

The Haskell library has wrapper functions for these kind of things, like `Command -> [String] -> String -> IO String`, where the `String` argument is the input to be sent to the process's stdin, and the result string is what the process printed to stdout. I didn't wrap those, though, because I figured we could just write them in unison.